### DISCUSSION TITLE
When using deferrable operators `task_instance` is saved with `start_date` equal to `end_date`

### DISCUSSION THREAD

**Discussion by jan920** [apache/airflow#21254] [General]
### Apache Airflow version

2.2.3 (latest released)

### What happened

When using deferrable operators `task_instance` doesn't have correct `start_date` and it is qual or close to `end_date` even when the task takes significant amount of time.

For example I used operator:

```
from airflow.sensors.time_delta import TimeDeltaSensorAsync
import datetime

job1 = TimeDeltaSensorAsync(delta=datetime.timedelta(seconds=120), task_id='job1')
```

During the run the `Gantt` was showing correct values on frontend: 
![Screenshot 2022-01-25 at 14 19 22](https://user-images.githubusercontent.com/38770318/150984950-406e1d2b-3a14-4b2c-b255-773cb2b7f882.png)

But after the run I can see that the `start_date` and `end_date` are the same.

![Screenshot 2022-01-25 at 14 20 23](https://user-images.githubusercontent.com/38770318/150985024-da5f894b-bc46-40b1-9af7-15d68a99a4a1.png)

![Screenshot 2022-01-25 at 14 20 05](https://user-images.githubusercontent.com/38770318/150985033-c4e93a53-d165-46c7-853c-164547d0516d.png)

### What you expected to happen

I think `start_date` should be equal to the actual datetime when the task started.

### How to reproduce

I was able to reproduce this by using following operator: 

```
from airflow.sensors.time_delta import TimeDeltaSensorAsync
import datetime

job1 = TimeDeltaSensorAsync(delta=datetime.timedelta(seconds=120), task_id='job1')
```

### Operating System

Darwin 21.1.0 x86_64

### Deployment

Virtualenv installation

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
I tihnk you need to explain a bit better what you mean. I read the description and looked at the screenshots few times and I do not understand what the problem is. I converted it into a discussion to clarify it.

**Comment by jan920** [comment]
Sorry. I'll try to describe it better. 

If `TimeDeltaSensorAsync(delta=datetime.timedelta(hours=1), task_id='job1')` starts at 8:00 1.2.2022 and finishes at 9:00 1.2.2022. It will have `start_date` and `end_date` equal to 9:00 1.2.2022. However, I would expect the `start_date` to be 8:00 1.2.2022 and `end_date` to be 9:00 1.2.2022. And this happens for all deferrable operators. 

Does this make more sense?

**Comment by jan920** [comment]
Actually my colleague found [this](https://github.com/apache/airflow/pull/20062) and it seems it should be fixing the issue.

But tanks a lot :)
